### PR TITLE
Fix panic if appchannel response is nil

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

* Fix panic if appchannel response is nil
* Add more test

## Issue reference

#1468 , #1409 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/grpc/api_test.go [hunk 1]
```diff
@@ -136,6 +140,23 @@ func startTestServer(port int) *grpc_go.Server {
...
+	server := grpc_go.NewServer()
+	go func() {
+		internalv1pb.RegisterDaprInternalServer(server, testAPIServer)
+		if err := server.Serve(lis); err != nil {
+			panic(err)
+		}
+	}()
+
+	// wait until server starts
+	time.Sleep(maxGRPCServerUptime)
```
Comment (yaron2) [inline pkg/grpc/api_test.go:155]: need to remove this, this will hurt cold start badly.

Comment (youngbupark) [reply pkg/grpc/api_test.go:155]: This is setting up the test server for testing. :)

Comment (yaron2) [reply pkg/grpc/api_test.go:155]: ah right, silly me, I thought it was the actual server

Comment (youngbupark) [reply pkg/grpc/api_test.go:155]: Actual code is just 3 lines, but I added 85 lines for test...

**Pull request merged**